ultralytics/trackers/utils/matching.py

```diff
@@ -34,19 +34,21 @@ def linear_assignment(cost_matrix, thresh, use_lap=True):
         return np.empty((0, 2), dtype=int), tuple(range(cost_matrix.shape[0])), tuple(range(cost_matrix.shape[1]))
 
     if use_lap:
+        # https://github.com/gatagat/lap
         _, x, y = lap.lapjv(cost_matrix, extend_cost=True, cost_limit=thresh)
-        matches = [[ix, mx] for ix, mx in enumerate(x) if mx >= 0]
+        matches = [[ix, mx] for ix, mx in enumerate(x) if mx >= 0 and cost_matrix[ix, mx] <= thresh]
```

Comment:
Hey I saw you updated the lap usage here. It was my understanding that the `cost_limit=thresh` argument in the line above already rejects matches below threshold. 

So does the new code actually change anything here or is it simply meant to align it closer to the scipy usage?

Reply:
I restore the changed line " and cost_matrix[ix, mx] <= thresh" and commit it.